src/Raven.Server/Documents/Patch/JavaScriptUtils.cs

```diff
@@ -49,8 +51,12 @@ public JavaScriptUtils(ScriptRunner runner, Engine engine)
         internal JsValue GetMetadata(JsValue self, JsValue[] args)
         {
             if (args.Length != 1 && args.Length != 2 || //length == 2 takes into account Query Arguments that can be added to args
-                !(args[0].AsObject() is BlittableObjectInstance boi)) 
+                !(args[0].AsObject() is BlittableObjectInstance boi))
+            {
+                if (args[0].AsObject() is GlobalObject) // Supposed to happen during the initialize of the script runner  - RavenDB-19466
+                    return JsValue.FromObject(_scriptEngine, EmptyMetadataDummy);
```

Comment:
Comment regarding this fix. When registering the behavior function in a script engine we pass _entire_ script (not only the definitions of behavior functions). This way we don't have to deal with extracting behavior functions. The drawback is that the regular script is still executed during the registration so we can detect parse errors:

https://github.com/ravendb/ravendb/blob/2f4167e983142b4fe60efa10c67e03b0052e4d15/src/Raven.Server/Documents/Patch/ScriptRunner.cs#L316-L323

 If the script contains `getMetadata(this)` then `this` is `GlobalObject` and we failed the check we have here.

The fix lets us bypass this validation (it's one time only) and move forward.